### DISCUSSION TITLE
PHP My Admin or adminer or pgadmin

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#358] [Ideas] [answered]
add function that the Nextcloud database can be opened from the Nextcloud AIO backend with PHP My Admin

**Comment by szaimen** [comment]
Hi, why do you need php my admin?

**Comment by ghost** [comment]
Nextcloud All in one is supposed to simplify the use and operation of Nextcloud for users and administrators.  Right?  As a Nextcloud administrator, I can say that it often happens that changes have to be made in the database (for example to remove inaccessible email accounts from the Nextcloud Mail app because this is not possible through the user interface or to reactivate blocked administrators because  they were accidentally deactivated due to incorrectly entered access data, ...) PHP My admin would make it easier to manage the database.

**Comment by szaimen** [comment]
Thanks for your good explanation.

On my personal instance I never had to use any database editing tools but I might have a small advantage here because I don't use the mail app. 

We are though not going to add this as default but will monitor if it should be requested more often and implement it in this case as optional addon.

**Comment by marcatsmci** [comment]
Please consider https://www.adminer.org/ as a PHPMyAdmin replacement : very light and usefull, MariaDB & PostgreSQl...
I use it for many years

**Comment by ghost** [comment]
also like this variant instead of phpmyadmin

**Comment by ghost** [comment]
> Thanks for your good explanation.
> 
> On my personal instance I never had to use any database editing tools but I might have a small advantage here because I don't use the mail app. 
> 
> We are though not going to add this as default but will monitor if it should be requested more often and implement it in this case as optional addon.

if a user has been blocked or an IP blocked because too many requests occur in the network, then you must have access to the database at the latest.

**Comment by rallisf1** [comment]
PhpMyAdmin won't do any good as this project uses postgres db. Adminer works but I find it a bit lacking. My personal preference is [pgadmin](https://hub.docker.com/r/dpage/pgadmin4/). As long as you attach the pgadmin container to the same network bridge as the nextcloud-AIO containers it connects just fine using postgres' container name for db hostname. Also remember to expose it to a different port than nextcloud, you can always use a reverse proxy to handle SSL for that later.

**Comment by marcatsmci** [comment]
I agree 🙂

Le 5 mars 2022 12:10:30 GMT+01:00, John Rallis ***@***.***> a écrit :
>PhpMyAdmin won't do any good as this project uses postgres db. Adminer works but I find it a bit lacking. My personal preference is [pgadmin](https://hub.docker.com/r/dpage/pgadmin4/). As long as you attach the pgadmin container to the same network bridge as the nextcloud-AIO containers it connects just fine using postgres' container name for db hostname. Also remember to expose it to a different port than nextcloud, you can always use a reverse proxy to handle SSL for that later.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/nextcloud/all-in-one/issues/106#issuecomment-1059744265
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Envoyé de mon appareil Android avec Courriel K-9 Mail.

**Comment by ghost** [comment]
I believe the inclusion of this feature is necessary. The following happened a few days ago on a non-AIO instance.

An instance with an administrator has the following setting:
Password Rules:
Number of login attempts before the user account is locked: 5

Due to a brute force attack, the number of login attempts before lockout was reached.

The user was disabled and could no longer be reached. The problem could be fixed by editing the corresponding entry in the database.
How can such a problem be solved on an AIO instance without access to the database?

**Reply by szaimen** [reply]
Simply use `sudo docker exec -it nextcloud-aio-nextcloud php occ security:bruteforce:reset [IP address]`?

See https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/occ_command.html#security

**Reply by szaimen** [reply]
Or rather `sudo docker exec -it nextcloud-aio-nextcloud php occ user:enable <name of user>`?

**Reply by ghost** [reply]
Ah nice, I actually didn't know about this possibility. Can this possibly be included in the documentation?

**Reply by szaimen** [reply]
yes

**Comment by szaimen** [comment, marked as answer]
Some documentation on this will be added with https://github.com/nextcloud/all-in-one/pull/850

**Comment by chiva** [comment]
@szaimen How can we use pgAdmin to connect to the database?

`docker run -p 8088:80 -e "PGADMIN_DEFAULT_EMAIL=user@domain.com" -e "PGADMIN_DEFAULT_PASSWORD=SuperSecret" dpage/pgadmin4 --network nextcloud-aio`

![CleanShot 2023-04-15 at 12 44 38](https://user-images.githubusercontent.com/305333/232209272-d421407c-eb0c-4697-bdf2-13fe8dbee4cc.png)

Using the container IP directly times out 🤷‍♂️

---------------------------------
Can you please add the docker command for *Adminer* to the Readme?
May not be the perfect one, but works like a charm and super easy to setup:
`docker run --network nextcloud-aio --link nextcloud-aio-database:db -p 8080:8080 adminer`

**Reply by tim-rohrer** [reply]
I'm also running into this bad connection message. @chiva what was your solution?

**Reply by chiva** [reply]
I recall using adminer, although adminer is now a deprecated project

**Reply by tim-rohrer** [reply]
Thanks for replying. I think my issue is docker related. For some reason, I'm starting the container but it isn't getting onto the `nextcloud-aio` network. I'm still trying to figure out why.

**Reply by derStephan** [reply]
> I recall using adminer, although adminer is now a deprecated project

It no longer is abandoned. There was quite some heavy work on it lately.
